src/documentMetadataManager.ts

```diff
@@ -28,64 +25,66 @@ export class DocumentMetadataManager {
     return DocumentMetadataManager.instance;
   }
 
+  private registerEventListeners() {
+    this.disposables.push(
+      workspace.onDidOpenTextDocument(this.handleDocumentOpen, this),
+      workspace.onDidSaveTextDocument(this.handleDocumentSave, this),
+      workspace.onDidCloseTextDocument(this.handleDocumentClose, this),
+      // TODO: Add rename/delete/move listeners as needed
+    );
+  }
+
+  /** Determines whether a document is relevant for tracking/logging */
+  private shouldLogDocumentEvent(document: TextDocument): boolean {
+    const { scheme } = document.uri;
+    const isCopilotVirtual = scheme === "vscode-chat-code-block";
+    return SUPPORTED_URI_SCHEMES.includes(scheme) && !isCopilotVirtual;
+  }
+
+  private logDocumentEvent(event: "opened" | "closed" | "saved", uri: Uri) {
+    logger.debug(`document ${event}`, { uri: uri.toString() });
+  }
```

Comment:
We can keep the existing logger lines as well - no need to wrap it in a new method since any additional listeners will also have to update the union type, which adds unnecessary maintenance